src/test/java/uk/gov/companieshouse/officer/delta/processor/tranformer/OfficerTransformTest.java

```diff
@@ -137,6 +137,32 @@ void onlyRolesWithDobIncludeDateOfBirth(OfficerRole officerRole) throws ProcessE
         }
     }
 
+    private static Stream<Arguments> emptyDobsWithDobRoles() {
+        Stream<OfficerRole> requiresDob = Arrays.stream(RolesWithDateOfBirth.values())
+                .map(RolesWithDateOfBirth::getOfficerRole);
+
+        return requiresDob.flatMap(role -> Stream.of(
+                Arguments.of(role, null),
+                Arguments.of(role, "")
+        ));
+    }
+
+    @DisplayName("Transformation doesn't fail when no DOB on role which requires it")
+    @ParameterizedTest
+    @MethodSource("emptyDobsWithDobRoles")
```

Comment:
May be simplified a bit using alternative
```
@Nullsource
@ValueSource(strings = {""})
```

Reply:
This would no be equivelant as the parameters are a cartesian product of Null and ValueSource but also of all of the RolesWithDateOfBirth Enum. The suggested changes will only inlcude the Null and ValueSource but not the enum. This could be considered sufficient. Does it need to check all roles? If not the suggested change would be sufficient.

Reply:
Sorry, you're right the test method has arguments (Role, String) not just (String).
Please ignore my previous comment, your code changes are good.